http/client.go

```diff
@@ -0,0 +1,68 @@
+package http
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+// NewWithCert creates a new http.Client with a custom certificate
+func NewWithCert(CABundleFile, CABundleDir string) (*http.Client, error) {
+	// go default http transport settings
+	transport := &http.Transport{
+		Proxy:                 http.ProxyFromEnvironment,
+		DialContext:           (&net.Dialer{Timeout: 30 * time.Second, KeepAlive: 30 * time.Second}).DialContext,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
+
+	if CABundleFile != "" || CABundleDir != "" {
+		certs, err := getCertPool(CABundleFile, CABundleDir)
+		if err != nil {
+			return nil, err
+		}
+		transport.TLSClientConfig = &tls.Config{RootCAs: certs}
+	}
+
+	return &http.Client{
+		Transport: transport,
+	}, nil
+}
+
+func getCertPool(certFile string, certDirectory string) (*x509.CertPool, error) {
+	caCertPool := x509.NewCertPool()
+	if certFile != "" {
+		caCert, err := ioutil.ReadFile(certFile)
+		if err != nil {
+			return nil, err
+		}
+
+		_ = caCertPool.AppendCertsFromPEM(caCert)
```

Comment:
Why are we not handling the returned value of these functions? If it isn't ok, probably we would like to debug some message, no?

Reply:
That func just returns if the cert was added to the pool

Reply:
Would you add a logger dep just to be able to know that in a debug?

Reply:
Done